sdk/openai/Azure.AI.OpenAI/src/Custom/SseReader.cs

```diff
@@ -0,0 +1,147 @@
+                if (!line.Value.IsComment)
+                    return line; // skip comment lines
+            }
+        }
+
+        // TODO: we should support cancellation tokens, but StreamReader does not in NS2
+        public async Task<SseLine?> TryReadSingleFieldEventAsync()
+        {
+            while (true)
+            {
+                var line = await TryReadLineAsync().ConfigureAwait(false);
+                if (line == null)
+                    return null;
+                if (line.Value.IsEmpty)
+                    throw new InvalidDataException("event expected.");
+                var empty = await TryReadLineAsync().ConfigureAwait(false);
+                if (empty != null && !empty.Value.IsEmpty)
+                    throw new NotSupportedException("Multi-filed events not supported.");
+                if (!line.Value.IsComment)
+                    return line; // skip comment lines
+            }
+        }
+
+        public SseLine? TryReadLine()
+        {
+            if (_reader.EndOfStream)
+                return null;
+            string lineText = _reader.ReadLine();
+            if (lineText.Length == 0)
+                return SseLine.Empty;
+            if (TryParseLine(lineText, out SseLine line))
+                return line;
+            return null;
+        }
+
+        // TODO: we should support cancellation tokens, but StreamReader does not in NS2
+        public async Task<SseLine?> TryReadLineAsync()
+        {
+            if (_reader.EndOfStream)
+                return null;
+            string lineText = await _reader.ReadLineAsync().ConfigureAwait(false);
+            if (lineText.Length == 0)
+                return SseLine.Empty;
+            if (TryParseLine(lineText, out SseLine line))
+                return line;
+            return null;
+        }
+
+        private static bool TryParseLine(string lineText, out SseLine line)
+        {
+            if (lineText.Length == 0)
+            {
+                line = default;
+                return false;
+            }
+
+            int colonIndex = lineText.IndexOf(':');
+
+            ReadOnlySpan<char> lineSpan = lineText.AsSpan();
+            ReadOnlySpan<char> fieldValue = lineSpan.Slice(colonIndex + 1);
+
+            bool hasSpace = false;
+            if (fieldValue.Length > 0 && fieldValue[0] == ' ')
+                hasSpace = true;
+            line = new SseLine(lineText, colonIndex, hasSpace);
+            return true;
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    _reader.Dispose();
+                    _stream.Dispose();
+                }
+
+                _disposedValue = true;
+            }
+        }
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+    }
+
+    public readonly struct SseLine
```

Comment:
should have its own file unless its private and only used by ssereader in which case lets make it private.

Reply:
Agreed. `SseLine` is part of the outward-facing contract, so I moved it into a new `SseLine.cs` file.